notifications/test_pipeline.py

```diff
@@ -30,5 +29,9 @@ def test_complete_pipeline(self):
         for i in history_list:
             self.assertIsInstance(i, PipelineHistory)
             self.assertIsInstance(i.timestamp, datetime)
+        # Check logging
+        self.assertEqual(len(mock_logging.info.call_args_list), 8)  # Successful delivery means 8 calls to logging info
+        self.assertEqual(len(mock_logging.warning.call_args_list), 0)
```

Comment:
Is it enough to simply print out the logging info and logging warning messages (if they exist)?